### ISSUE TITLE
Focus state has insufficient contrast over light colors

### ISSUE THREAD

**Issue by ericronne** [18F/calc#1740]
Update the focus color to the following specs:
- `$color-gray` for links which run over **white**
- `$color-gray-darker` for links which run over light colors (e.g. green in footer)
- `$color-gray-light` for links which run over dark colors (e.g. header)

**Current footer**
![image](https://user-images.githubusercontent.com/9259185/38574095-91efa2a2-3cbd-11e8-9069-2a56f189505e.png)

**Current site admin page**
![image](https://user-images.githubusercontent.com/9259185/38574155-bebe8e1a-3cbd-11e8-8daf-946559bca8e6.png)

**Comment by toolness** [comment]
Oh I was about to file a new issue but I found this one. Here's another case we'll want to make sure to cover:

![context-menu-color](https://user-images.githubusercontent.com/124687/39704701-288fbda6-51da-11e8-967e-871e98ac21cc.png)

**Comment by ericronne** [comment]
Yep. Even USWDS suffers from this issue. I believe they use one color for all situations. Contrast is sufficient over darks, but not over white:
![image](https://user-images.githubusercontent.com/9259185/39707548-60106b72-51da-11e8-988d-76c1d3167d6c.png)

I don't think a one-color solution is possible. :/

**Comment by toolness** [comment]
DOGGONE IT RONNE

hmm.  if the user has javascript enabled we might be able to do some funky thing that dynamically sets the focus state based on the perceived background colour.  if it's hard to write a stylesheet to do that kind of thing for us.

**Comment by ericronne** [comment]
it's beyond my ken, at any rate

@hbillings ?

**Comment by hbillings** [comment]
We can do this with CSS mixins, I believe. We'll need to define which color backgrounds need which color borders and create a li'l CSS "kit" for them. Then we can do a bit like WDS does it with headlines and links in the `usa-section-dark` class (except I'd rather do it with mixins, because then we don't need to mess with the DOM _and_ it'll be less likely that we end up applying classes willy-nilly to try to get things to look right).

**Comment by ericronne** [comment]
That sounds amazing! Would love to flywall if we do that. Should we make an issue?

**Comment by hbillings** [comment]
Sounds good! This issue is just fine; you can assign it to me and you. :)

**Comment by ericronne** [comment]
## currently only use a single color: $color-gray-light
![image](https://user-images.githubusercontent.com/9259185/39763213-23b7475e-52a2-11e8-91c0-71ff8fcd6960.png)

## let’s use two:
**over darks:** `$color-gray-light`
**over lights:** `$color-gray`
![image](https://user-images.githubusercontent.com/9259185/39763320-67281248-52a2-11e8-88f8-4ba05aee8c2d.png)
